ts/index.ts

```diff
@@ -0,0 +1,193 @@
+    // It `extra-tags` is also not set, which means that it's an empty string, that's
+    // still valid, as the flakehub-push CLI expects a comma-separated list here.
+    return labels !== "" ? labels : tags;
+  }
+
+  // We first check for a value using the `source-binary` input and fall back to the
+  // now-deprecated `flakehub-push-binary`
+  private get sourceBinary(): string | null {
+    const sourceBinaryInput = inputs.getStringOrNull("source-binary");
+    const flakeHubPushBinaryInput = inputs.getStringOrNull(
+      "flakehub-push-binary",
+    );
+
+    return sourceBinaryInput !== ""
+      ? sourceBinaryInput
+      : flakeHubPushBinaryInput;
+  }
+
+  private get visibility(): Visibility {
+    const visibility = inputs.getString("visibility");
+    if (!VISIBILITY_OPTIONS.includes(visibility)) {
+      actionsCore.setFailed(
+        `Visibility option \`${visibility}\` not recognized. Available options: ${VISIBILITY_OPTIONS.map((opt) => `\`${opt}\``).join(", ")}.`,
+      );
+    }
+    return visibility as Visibility;
+  }
+
+  private async executionEnvironment(): Promise<ExecutionEnvironment> {
+    const env: ExecutionEnvironment = {};
+
+    env.FLAKEHUB_PUSH_VISIBILITY = this.visibility;
+    env.FLAKEHUB_PUSH_TAG = this.tag;
+    env.FLAKEHUB_PUSH_HOST = this.host;
+    env.FLAKEHUB_PUSH_LOG_DIRECTIVES = this.logDirectives;
+    env.FLAKEHUB_PUSH_LOGGER = this.logger;
+    env.FLAKEHUB_PUSH_GITHUB_TOKEN = this.gitHubToken;
+    env.FLAKEHUB_PUSH_REPOSITORY = this.repository;
+    env.FLAKEHUB_PUSH_DIRECTORY = this.directory;
+    env.FLAKEHUB_PUSH_GIT_ROOT = this.gitRoot;
+    // not included: the now-deprecated FLAKEHUB_PUSH_EXTRA_TAGS
+    env.FLAKEHUB_PUSH_EXTRA_LABELS = this.extraLabels;
+    env.FLAKEHUB_PUSH_SPDX_EXPRESSION = this.spdxExpression;
+    env.FLAKEHUB_PUSH_ERROR_ON_CONFLICT = this.errorOnConflict.toString();
+    env.FLAKEHUB_PUSH_INCLUDE_OUTPUT_PATHS = this.includeOutputPaths.toString();
+    env.FLAKEHUB_PUSH_ROLLING = this.rolling.toString();
+    env.FLAKEHUB_PUSH_MIRROR = this.mirror.toString();
+
+    if (this.name !== null) {
+      env.FLAKEHUB_PUSH_NAME = this.name;
+    }
+
+    if (this.rollingMinor !== null) {
+      env.FLAKEHUB_PUSH_ROLLING_MINOR = this.rollingMinor.toString();
+    }
+
+    return env;
+  }
+
+  async push(): Promise<void> {
+    const executionEnv = await this.executionEnvironment();
+
+    const binary =
+      this.sourceBinary !== null
+        ? this.sourceBinary
+        : await this.idslib.fetchExecutable();
+
+    actionsCore.debug(
+      `execution environment: ${JSON.stringify(executionEnv, null, 2)}`,
+    );
+
+    const exitCode = await actionsExec.exec(
+      binary,
+      // We're setting this via flag for now due to a misspelling in the original environment variable.
+      // Remove this in favor of the environment variable only after PR #125 is merged.
+      ["--visibility", this.visibility],
```

Comment:
Should we comment out the env above for the moment then? So that it's easier to reason about which one is actually being chosen?